File: Data Structures & Algorithms/remove-element/submission-0.cpp
class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        int new_size = 0;
        for (int i = 0; i< nums.size(); i++) {
            if (nums[i] != val) {
                nums[new_size] = nums[i];
                new_size++;
            }
        }
        return new_size;
    }
};

File: Data Structures & Algorithms/majority-element/submission-0.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    majorityElement(nums) {
        const hash_map = new Map();
        nums.forEach((eachNumber) => {
            if (hash_map.has(eachNumber)) {
                hash_map.set(eachNumber,hash_map.get(eachNumber) + 1)
            } else {
                hash_map.set(eachNumber, 1);
            }
        });

        let parScore = nums.length / 2;
        let result;
        hash_map.forEach((value, key) => {
            if (value > parScore) {
                result =  key;
            }
        })
        return result;
    }
}
